service/index.go

```diff
@@ -1,21 +1,65 @@
 package service
 
 import "github.com/blevesearch/bleve"
+import "log"
 
 // IndexService specifies an API to interact with indexes.
 type IndexService interface {
-	Create(path string) (bleve.Index, error)
+	Create(indexName string, id string, data interface{}) (bleve.Index, error)
+	Get(indexName string, id string) (interface{}, error)
 }
 
 // DefaultIndexService is a default implementation of IndexService using bleve.
 type DefaultIndexService struct{}
 
 // Create creates an index.
-func (*DefaultIndexService) Create(path string) (bleve.Index, error) {
-	mapping := bleve.NewIndexMapping()
-	index, err := bleve.New(path, mapping)
+func (*DefaultIndexService) Create(indexName string, id string, data interface{}) (bleve.Index, error) {
+	//TODO: Don't open and close the index for every request
+	index, err := openOrCreateIndex(indexName)
 	if err != nil {
 		return nil, err
 	}
+	defer index.Close()
+
+	if err := index.Index(id, data); err != nil {
+		return nil, err
+	}
+
+	return index, nil
+}
+
+// Get returns a document with id `id`
+func (*DefaultIndexService) Get(indexName string, id string) (interface{}, error) {
+	//TODO: Don't open and close the index for every request
+	index, err := openOrCreateIndex(indexName)
```

Comment:
I don't agree with nonexistent indices being created upon attempting to query them.